bin/civihr-install

```diff
@@ -0,0 +1,808 @@
+function civicrm_make_setup_conf() {
+  civihr_assertvars civicrm_make_setup_conf PRJDIR CMS_ROOT CIVI_CORE CIVI_UF CIVI_DB_NAME CIVI_DB_USER CIVI_DB_PASS
+
+  cat > "$CIVI_CORE/bin/setup.conf" << EOF
+    PRJDIR="$PRJDIR"
+    CMS_ROOT="$CMS_ROOT"
+    DBNAME="$CIVI_DB_NAME"
+    DBUSER="$CIVI_DB_USER"
+    DBPASS="$CIVI_DB_PASS"
+    DBHOST="$CIVI_DB_HOST"
+    DBPORT="$CIVI_DB_PORT"
+    SVNROOT="$CIVI_CORE"
+    CIVISOURCEDIR="$CIVI_CORE"
+    SCHEMA=schema/Schema.xml
+    DBARGS=""
+    PHP5PATH=""
+    DBLOAD="$DBLOAD"
+    GENCODE_CMS="$CIVI_UF"
+EOF
+}
+
+###############################################################################
+## Drupal -- Generate config files and setup database
+function drupal_install() {
+  drupal7_install
+}
+
+###############################################################################
+## Drupal -- Generate config files and setup database
+function drupal7_install() {
+  civihr_assertvars drupal7_install CMS_ROOT SITE_ID CMS_TITLE CMS_DB_USER CMS_DB_PASS CMS_DB_HOST CMS_DB_NAME ADMIN_USER ADMIN_PASS CMS_URL
+
+  DRUPAL_SITE_DIR="$SITE_ID"
+
+  CMS_DB_HOSTPORT=$(civihr_build_hostport "$CMS_DB_HOST" "$CMS_DB_PORT")
+
+  pushd "$CMS_ROOT" >> /dev/null
+    [ -f "sites/$DRUPAL_SITE_DIR/settings.php" ] && rm -f "sites/$DRUPAL_SITE_DIR/settings.php"
+
+    drush site-install -y "$@" \
+      --db-url="mysql://${CMS_DB_USER}:${CMS_DB_PASS}@${CMS_DB_HOSTPORT}/${CMS_DB_NAME}" \
+      --account-name="$ADMIN_USER" \
+      --account-pass="$ADMIN_PASS" \
+      --account-mail="$ADMIN_EMAIL" \
+      --site-name="$CMS_TITLE" \
+      --sites-subdir="$DRUPAL_SITE_DIR"
+    chmod u+w "sites/$DRUPAL_SITE_DIR"
+    chmod u+w "sites/$DRUPAL_SITE_DIR/settings.php"
+    chmod u-w "sites/$DRUPAL_SITE_DIR/settings.php"
+
+    ## Setup extra directories
+    civihr_mkdir "sites/${DRUPAL_SITE_DIR}/files"
+    civihr_mkdir "sites/${DRUPAL_SITE_DIR}/modules"
+  
+  popd >> /dev/null
+}
+
+###############################################################################
+## Initialize (or update) a cached copy of a git repo in $CACHE_DIR
+## usage: git_cache_setup <url> <cache-dir>
+function git_cache_setup() {
+  local url="$1"
+  local cachedir="$2"
+  local lock="${cachedir}.lock"
+  local lastrun="${cachedir}.lastrun"
+  ## TODO: defensive programming: $cachedir should not end in "/"
+
+  if [ -d "$cachedir" -a -f "$lastrun" -a -z "$FORCE_DOWNLOAD" ]; then
+    if php -r "exit($argv[1] + file_get_contents($argv[2]) < time() ? 1 : 0);" -- $CACHE_TTL "$lastrun" ; then
+      echo "SKIP: git_caccaches.shhe_setup '$url' $cachedir' (recently updated; ttl=$CACHE_TTL)"
+      return
+    fi
+  fi
+
+  civihr_makeparent "$lock"
+  if pidlockfile.php "$lock" $$ $CACHE_LOCK_WAIT ; then
+    php -r 'echo time();' > "$lastrun"
+    if [ ! -d "$cachedir" ]; then
+      ## clone
+      echo "[[Initialize cache dir: $cachedir]]"
+      civihr_makeparent "$cachedir"
+      timeout.php $SCM_TIMEOUT git clone --mirror "$url" "$cachedir"
```

Comment:
Instead of cloning the repository, which makes mandatory to have git installed, can't we download the tarballs from github? Github provides those nice URLs where you can get them. For example, the tarball for civihr staging can be downloaded at https://github.com/civicrm/civihr/archive/staging.zip